src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayMissingNameAndRefTagCheck.java

```diff
@@ -0,0 +1,376 @@
+    }
+
+    /**
+     * Function to collect osm name tags
+     * 
+     * @param edges
+     *            a set of edges (connected edges from edge in question)
+     * @return set of strings representing name tags
+     */
+    private Set<String> collectEdgeNameTags(final Set<Edge> edges)
+    {
+        final Set<String> edgeNames = new HashSet<>();
+        for (final Edge edge : edges)
+        {
+            final Map<String, String> tags = edge.getTags();
+            if (tags.containsKey(NameTag.KEY))
+            {
+                edgeNames.add(tags.get(NameTag.KEY));
+            }
+        }
+        return edgeNames;
+    }
+
+    /**
+     * Function to collect osm reference tags
+     * 
+     * @param edges
+     *            a set of edges (connected edges from edge in question)
+     * @return Set of strings representing reference tags
+     */
+    private Set<String> collectEdgeRefTags(final Set<Edge> edges)
+    {
+        final Set<String> edgeRefs = new HashSet<>();
+        for (final Edge edge : edges)
+        {
+            final Map<String, String> tags = edge.getTags();
+            if (tags.containsKey(ReferenceTag.KEY))
+            {
+                edgeRefs.add(tags.get(ReferenceTag.KEY));
+            }
+        }
+        return edgeRefs;
+    }
+
+    /**
+     * Function to determine if there are common tags between edges (helper function to ignore small
+     * connector road)
+     *
+     * @param tagSet1
+     *            start/end node connected edges set of tags
+     * @param tagSet2
+     *            start/end node connected edges set of tags
+     * @return boolean if there are common tags between edges.
+     */
+    private boolean edgesShareTags(final Set<String> tagSet1, final Set<String> tagSet2)
+    {
+        for (final String tag : tagSet1)
+        {
+            if (tagSet2.contains(tag))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Function to determine if Edge object is missing both name and ref tag
+     * 
+     * @param tags
+     *            object tags
+     * @return boolean if Edge does not contain either name or ref tag.
+     */
+    private boolean highwayMissingBothNameAndRefTag(final Map<String, String> tags)
+    {
+        return !tags.containsKey(NameTag.KEY.toLowerCase())
+                && !tags.containsKey(ReferenceTag.KEY.toLowerCase());
+    }
+
+    /**
+     * function to determine which inEdge's name/ref tags to add to respective sets if applicable
+     * based on difference in angle between edge in question and inEdge.
+     *
+     * @param edge
+     *            edge in question
+     * @param inEdges
+     *            edge in question's inEdges
+     * @param inEdgeNames
+     *            Set of inEdges' names
+     * @param inEdgeRefs
+     *            Set of inEdges' refs
+     */
+    private void inEdgeLogic(final Edge edge, final Set<Edge> inEdges,
```

Comment:
Couldn't this be combined with outEdgeLogic?

Reply:
I originally had them together but the cognitive complexity was 23

Reply:
yeah I will put this into one function, just need to account for inEdge vs outEdge for the angleDifference between edges.